orders-accounting/src/storage/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{
+    chain::{OrderData, OrderId},
+    primitives::Amount,
+};
+use std::ops::{Deref, DerefMut};
+
+pub mod db;
+pub mod in_memory;
+
+pub trait OrdersAccountingStorageRead {
+    type Error: std::error::Error;
+
+    fn get_order_data(&self, id: &OrderId) -> Result<Option<OrderData>, Self::Error>;
+    fn get_ask_balance(&self, id: &OrderId) -> Result<Option<Amount>, Self::Error>;
+    fn get_give_balance(&self, id: &OrderId) -> Result<Option<Amount>, Self::Error>;
```

Comment:
The wording here is very confusing. Usually it's ask/bid, not ask/give. Not sure whether you have a reason for a different naming.

Reply:
So my idea was to represent order as an exchange of OutputValue types: you ask for x amount of A and gives y amount of B in return. This looks both universal and precise (i.e. avoids storing price).
Imo ask/bid are more applicable for trading with buy/sell orders that both parties can submit.